### PR TITLE
Do not compile unless requested.

### LINKED ISSUE #5505: Make include resolution blazing fast
There should be an option to perform "include resolution only" - it might be that not selecting any output could already do that, but I think it will still do type checking. It might also be that if any include fails, we do not do anything at all.

Task of this issue:
 - check that truffle (and other framework)'s include resolution is as fast as it gets.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes #5505

### CODE REVIEW

File: libsolidity/interface/CompilerStack.cpp [hunk 1]
```diff
@@ -389,18 +389,27 @@ string const CompilerStack::lastContractName() const
 
 eth::AssemblyItems const* CompilerStack::assemblyItems(string const& _contractName) const
 {
+	if (m_stackState != CompilationSuccessful)
+		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Compilation was not successful."));
+
 	Contract const& currentContract = contract(_contractName);
```
Comment (axic) [inline libsolidity/interface/CompilerStack.cpp:395]: IIRC `contract` does this check already.

Comment (chriseth) [reply libsolidity/interface/CompilerStack.cpp:395]: still, wanted to play safe.

**Review by ekpyron — DISMISSED** [review]
Looks good! Should probably have a changelog entry, though, since in some cases (like stack height issues during code generation) this will behave differently than before.

**Comment by chriseth** [comment]
It also fails one of the tests.... :)

**Review by leonardoalt — DISMISSED** [review]

**Comment by axic** [comment]
To ease reviewing and following up changes, this should have been 3 PRs:
1) not emitting empty objects in StandardCompiler
2) safety assertions in CompilerStack
3) not running compile if it is not requested

I didn't fully review the safety assertions, but some of them seem to change semantics. They are done twice - once in the caller and once in `contracts()`, which could be removed. Also having a helper for the assertion may be beneficial at this stage to avoid that much code duplication.

OK merging this if there's no time to address these concerns. It seems to be ok.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**